epyqlib/updateepc.py

```diff
@@ -486,11 +488,16 @@ def version(path):
     except:
         pass
 
-    ui_files = tuple(
-        f
-        for f in referenced_files
-        if f.endswith('.ui')
-    )
+    # path might be a string or a pathlib.Path object
+    # need to be able to handle both 
+    ui_files = []
+    for f in referenced_files:
+        if isinstance(f, pathlib.Path):
+            if f.name.endswith('.ui'):
+                ui_files.append(f.name)
+        else:
+            if f.endswith('.ui'):
+                ui_files.append(f)
```

Comment:
Let's make the paths consistent pre-emptively instead of conditionally handling both `str` and `pathlib.Path`.  I think it would be more proper (than what I put below) to actually correct the types at the sources but that grows a bit big to just tack into this PR.  In general I ought to take a pass at removing `str` paths _everywhere_...  but that is a separate task.

```python
referenced_files = tuple(pathlib.Path(path) for path in referenced_files)

ui_files = tuple(
    f
    for f in referenced_files
    if f.suffix == '.ui'
)
```

On the topic of `tuple()` vs. `[]`...  I'm really not sure where I stand.  So, I wrote it one way here instead of debating endlessly with myself.  Basically I am trying to tend towards immutable (tuple) but a list just writes and reads so much nicer.  bah.

Reply:
slick, I didn't realize if you passed a pathlib.Path object to pathlib.Path() it just returned the object. Will update.

Reply:
It doesn't, it does create a new one of the same 'value'.  But, you make sure you have all the same type this way.  I suppose you could do as below but I don't think the maybe-optimization is worth the readability cost (until we identify it is a real performance problem).

```python
referenced_files = tuple(
    path if isinstance(path, pathlib.Path) else pathlib.Path(path)
    for path in referenced_files
)
```

Reply:
I see, good to know. Leaving as is works for me.